### PR TITLE
Offline Mode for zingo-mobile

### PR DESCRIPTION
**Pull request by dorianvp**

### CODE REVIEW

**Comment by zancas** [comment]
@dorianvp can this be tested with Kotlin in zingo-mobile?  Or detox?

File: zingolib/src/config.rs [hunk 1]
```diff
@@ -98,6 +98,45 @@ pub fn load_clientconfig(
...
+pub fn load_clientconfig_serverless(
+    lightwallet_uri: http::Uri,
+    data_dir: Option<PathBuf>,
+    chain: ChainType,
+    monitor_mempool: bool,
+) -> std::io::Result<ZingoConfig> {
+    use std::net::ToSocketAddrs;
+    match format!(
+        "{}:{}",
+        lightwallet_uri.host().unwrap(),
```
Comment (zancas) [inline zingolib/src/config.rs:76]: Who's passing this uri to this function?   Should we add a error enum to handle the failures that can occur inside this fn?

Probably using all the functionality that `thiserror` offers to be ergonomic in the definition?

Comment (zancas) [reply zingolib/src/config.rs:76]: Given that the parameter is a Uri, how can host() and port() fail?

Can we write unit-tests that force the exercise of these failure modes?

Comment (dorianvp) [reply zingolib/src/config.rs:76]: The ideal thing would be to alter the `construct_lightwalletd_uri` method to return a `Result<http::Uri, http::uri::InvalidUri>`. We could use `panic::catch_unwind` meanwhile in tests, and then have another PR with the refactor in place.

Comment (zancas) [reply zingolib/src/config.rs:76]: I agree let's change the return of `construct_lightwalletd_uri` to the `Result` suggested.

Comment (zancas) [reply zingolib/src/config.rs:76]: We can do this later.

**Review by zancas — APPROVED** [review]

File: zingolib/src/config.rs [hunk 2]
```diff
@@ -726,3 +730,42 @@ impl ActivationHeights {
...
+    #[tokio::test]
+    async fn test_load_clientconfig_serverless() {
+        rustls::crypto::ring::default_provider()
+            .install_default()
+            .expect("Ring to work as a default");
+        tracing_subscriber::fmt().init();
+
+        let valid_uri = crate::config::construct_lightwalletd_uri(Some(
+            crate::config::DEFAULT_LIGHTWALLETD_SERVER.to_string(),
+        ));
```
Comment (juanky201271) [inline zingolib/src/config.rs:745]: I don't really think this test is correct, I see `valid_uri` with a valid uri, hehehehe, obviously, and IMO this test needs a different value here, IDK like `http::Uri::default()` which for me is a `empty` server uri.

File: zingolib/src/config.rs [hunk 3]
```diff
@@ -726,3 +730,42 @@ impl ActivationHeights {
...
+
+        let valid_config = crate::config::load_clientconfig(
+            valid_uri.clone(),
+            Some(temp_path),
+            crate::config::ChainType::Mainnet,
+            true,
+        )
+        .unwrap();
+
+        assert_eq!(valid_config.get_lightwalletd_uri(), valid_uri);
```
Comment (juanky201271) [inline zingolib/src/config.rs:760]: this make no sense to me... in a serverless scenario... in fact the normal situation for `serverless` is:
- server uri: empty.
- chain type: empty.

**Pull request merged**